common/cards/card-plugins/effects/diamond-armor.js

```diff
@@ -1,30 +1,66 @@
-import {AttackModel} from '../../../../server/models/attack-model'
 import {GameModel} from '../../../../server/models/game-model'
 import EffectCard from './_effect-card'
 
+/**
+ * @typedef {import('common/types/cards').CardPos} CardPos
+ */
+
 class DiamondArmorEffectCard extends EffectCard {
 	constructor() {
 		super({
 			id: 'diamond_armor',
 			name: 'Diamond Armor',
 			rarity: 'rare',
 			description:
-				'Protects from the first +30hp damage.\n\nDiscard after user is knocked out.',
+				'Prevent up to 20hp damage taken.\n\nAlso prevents additional damage to this Hermit from all effect cards.',
 		})
 	}
 
 	/**
-	 *
 	 * @param {GameModel} game
 	 * @param {string} instance
-	 * @param {AttackModel} attack
+	 * @param {CardPos} pos
 	 */
-	onDefence(game, instance, attack) {
-		if (attack.type !== 'ailment') {
-			attack.defence.damageReduction += 30
+	onAttach(game, instance, pos) {
+		const {otherPlayer, player} = pos
+		const instanceKey = this.getInstanceKey(instance)
+
+		otherPlayer.hooks.onAttack[instance] = (attack, pickedSlots) => {
+			if (attack.target.index !== pos.rowIndex || attack.type === 'ailment')
```

Comment:
no need to filter out ailment, you are filtering for effects

Reply:
Not needed for the Chainmail Armor but it's needed for the Diamond and Netherite armor, they block up to 20 damage from other sources.